plugins/modules/elastic_role_mapping.py

```diff
@@ -69,7 +69,8 @@ def main():
     enabled=dict(type='bool', default=True),
     roles=dict(type='list', required=True),
     rules=dict(type='dict', required=True),
-    metadata=dict(type='dict', default={})
+    metadata=dict(type='dict', default={}),
+    deployment_info=dict(type='dict', default=None)
```

Comment:
not a big deal, but, you don't need to default to `None` on  inputs, that will happen automatically.